src/cursor/cur_table.c

```diff
@@ -433,22 +433,22 @@ __curtable_prev(WT_CURSOR *cursor)
 static int
 __curtable_reset(WT_CURSOR *cursor)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i;
 
     ctable = (WT_CURSOR_TABLE *)cursor;
-    /* Grab the primary cursor to reset the bounds. */
-    primary = *ctable->cg_cursors;
 
     JOINABLE_CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, reset, NULL);
 
     APPLY_CG(ctable, reset);
 
-    /* If a user calls cursor reset also reset the bounds. */
+    /* If a user calls cursor reset, call cursor reset on all column groups. */
```

Comment:
The `API_USER_ENTRY` macro was added to prevent internal API calls from resetting the cursor bounds while still allowing `__curfile_reset` to reset cursor bounds. In certain cases particularly the table cursor `APPLY_CG(ctable, reset)` is called for write operations e.g. curtable_remove. To prevent confusion about whether a call would result in a bounds being removed from the cursor we decided to add the restriction that only an external call to `cursor->reset` should clear bounds. This then results in what appears to be redundant code in `__curtable_reset` as to work around it we need to call bounds clear on the column group cursors themselves directly.

Its an annoying semantic, either we unintentionally reset occasionally which could be confusing for the user or we have to maintain a bit of extra logic when we reset bounds on cursor groups.

Reply:
I will add a large comment explaining this.

Reply:
[7ef3190](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/7ef3190b61449578ace49126ccf4c268948584fa)